tests/e2e/monovertex.go

```diff
@@ -73,21 +73,31 @@ func VerifyMonoVertexRolloutReady(monoVertexRolloutName string) {
 	}).Should(Equal(metav1.ConditionTrue))
 }
 
-func VerifyMonoVertexReady(namespace, monoVertexRolloutName string) {
+func VerifyMonoVertexReady(namespace, monoVertexRolloutName string) error {
 
 	By("Verifying that the MonoVertex is running")
 	monoVertexName := VerifyMonoVertexStatus(namespace, monoVertexRolloutName,
 		func(retrievedMonoVertexSpec numaflowv1.MonoVertexSpec, retrievedMonoVertexStatus kubernetes.GenericStatus) bool {
 			return retrievedMonoVertexStatus.Phase == string(numaflowv1.MonoVertexPhaseRunning)
 		})
 
-	vertexLabelSelector := fmt.Sprintf("%s=%s,%s=%s", numaflowv1.KeyMonoVertexName, monoVertexName, numaflowv1.KeyComponent, "mono-vertex")
-	daemonLabelSelector := fmt.Sprintf("%s=%s,%s=%s", numaflowv1.KeyMonoVertexName, monoVertexName, numaflowv1.KeyComponent, "mono-vertex-daemon")
+	unstructMonoVertex, err := GetMonoVertex(namespace, monoVertexRolloutName)
```

Comment:
We may want to add a comment to this function that says when calling this we should already have verified that the "promoted" one has been updated before getting here. The thing is that calling `GetMonoVertex()` basically finds the "promoted" one. So, if we haven't updated the "promoted" one yet, then that would find the original one.

Ultimately, it may be better to use the spec from the Rollout rather than the spec from the child in all of the functions and try to remove all of these calls to `GetMonoVertex()`, `GetPipeline()`, etc, but I'm fine if we create an issue for someone to do later.

Reply:
(I know the existing code already had this problem btw)

Reply:
Added [issue](https://github.com/numaproj/numaplane/issues/612) - we can do it at some point